research/2019-assets/add-to-calendar/add-to-calendar-20201102.js

```diff
@@ -0,0 +1,173 @@
+							prop_val_cache = prop_cache.innerText;
+							googleLink += "&text=" + prop_val_cache;
+							icsLink += "\nSUMMARY:" + prop_val_cache;
+						}
+						break;
+					case "description":
+						prop_val_cache = prop_cache.innerText;
+						googleLink += "&details=" + prop_val_cache;
+						icsLink += "\nDESCRIPTION:" + prop_val_cache.replace( /(\r\n|\n|\r)/gm, " " );
+						break;
+					case "startDate":
+						prop_val_cache = prop_cache.getAttribute( "content" ).replace(/(-|:)/g, "");
+						googleLink += "&dates=" + prop_val_cache;
+						icsLink += "\nUID:" + uid + "-" + prop_val_cache + "\nDTSTART:" + prop_val_cache;
+						break;
+					case "endDate":
+						prop_val_cache = prop_cache.getAttribute( "content" );
+						googleLink += "/" + prop_val_cache.replace(/(-|:)/g, "");
+						icsLink += "\nDTEND:" + prop_val_cache.replace(/(-|:)/g, "");
+						break;
+					case "streetAddress":
+						prop_val_cache = prop_cache.innerText;
+						googleLink += " " + prop_val_cache;
+						icsLink += " " + prop_val_cache;
+						break;
+					case "addressLocality":
+						prop_val_cache = prop_cache.innerText;
+						googleLink += " " + prop_val_cache;
+						icsLink += " " + prop_val_cache;
+						break;
+					case "addressRegion":
+						prop_val_cache = prop_cache.innerText;
+						googleLink += " " + prop_val_cache;
+						icsLink += " " + prop_val_cache;
+						break;
+					case "postalCode":
+						prop_val_cache = prop_cache.innerText;
+						googleLink += " " + prop_val_cache;
+						icsLink += " " + prop_val_cache;
+						break;
+				}
+			}
+			
+			if ( !place ) {
+				place = elm.querySelector( "." + componentName + "-location" ).innerText;
+				googleLink += "&location=" + place;
+				icsLink += "\nLOCATION:" + place;
+			}
+			
+			alert(test2);
+			
+			googleLink = encodeURI(googleLink);
+			icsLink = icsLink + "\nEND:VEVENT\nEND:VCALENDAR";
+
+			//Create add details summary to the wb-addcal event and initiate the unordered list
+			$elm.append("<details class='max-content " + componentName + "-buttons'><summary>" + i18nDict.addto + " " + i18nDict.calendar +  "</summary><ul class='list-unstyled mrgn-bttm-0 mrgn-tp-sm'><li><a class='btn btn-link btn-lg mrgn-top-lg' href='" + googleLink + "'><span class='fab fa-google mrgn-rght-md'></span>Google<span class='sr-only'>" + i18nDict.calendar +"</span></a></li><li><button class='btn btn-link btn-lg' data-" + componentName + "-ics='" + icsLink + "'><span class='fa fa-calendar mrgn-rght-md'></span>" + i18nDict.ical + "<span class='sr-only'>Calendar</span></button></li>");
+				
+			//Set click action on Apple and Other button links
+			var linkItem = elm.querySelectorAll( "." + componentName + "-buttons button" );
+			i_len = linkItem.length;
+			
+			for ( i=0; i < i_len; i++ ) {
+				linkItem[ i ].onclick = function() {
+					wb.download( new Blob( [ this.getAttribute( "data-" + componentName +"-ics" ) ] , { type: "text/calendar;charset=utf-8" } ), "evenement-gc-event.ics" );
```

Comment:
We would need to allow the filename to be configurable, and the reference "-gc-" will need to be removed